ctlearn/image_mapping.py

```diff
@@ -840,20 +829,16 @@ def get_virtual_pixels(self, x_ticks, y_ticks, x, y):
         return virtual_pixels
 
     def normalization(self, mapping_matrix3d, num_pixels):
-        norm_factor = 0
-        for i in np.arange(1,mapping_matrix3d.shape[0],1):
-            norm_factor += np.sum(mapping_matrix3d[i])
+        norm_factor = np.sum(mapping_matrix3d[1:])
         norm_factor /= float(num_pixels)
-        for i in np.arange(1,mapping_matrix3d.shape[0],1):
-            mapping_matrix3d[i] /= norm_factor
+        mapping_matrix3d[1:] /= norm_factor
         return mapping_matrix3d
                                                        
     def mask_interpolation(self, mapping_matrix3d, nn_index, num_pixels, pad):
         for i in np.arange(0,nn_index.shape[0],1):
             for j in np.arange(0,nn_index.shape[1],1):
                 if nn_index[j][i] >= num_pixels:
-                    for k in np.arange(1,mapping_matrix3d.shape[0],1):
-                        mapping_matrix3d[k][j+pad][i+pad] = 0.0
+                    mapping_matrix3d[1:][j+pad][i+pad] = 0.0
```

Comment:
This line is causing the holes in the images, if mask bilinear and mask bicubic interpolation is selected. The slice assignment can't operate as a for loop. I did a quite test, where we want to set all second elements to zero (similar to the mapping tables). But the slice ':' is setting the whole second array (similar to one mapping table) to zero.
![screen shot 2019-02-13 at 10 35 34](https://user-images.githubusercontent.com/37835610/52701717-2b924580-2f7b-11e9-81b5-f8f16c44491c.png)

I thought on calculating a mask (2D array) before and then multiplying the mask to all mapping tables. The k loop would be moved outside of the i and j loop, so we will ensure the fast runtime. I don't know why i didn't do it in the first place!